### ISSUE TITLE
Separate show rules for text and math

### ISSUE THREAD

**Issue by knuesel** [typst/typst#4137]
### Description

Sometimes the same element in text vs math is semantically different and it would be helpful to have a show rule apply in text but not in math, or the other way.

### Use Case

For example I might want to treat `?` as a normal character in math mode to avoid the extra spacing:

```
Point $P(1, ?)$   // ugly space after the `?`
```

![image](https://github.com/typst/typst/assets/2412819/7858fc0b-4cea-48ca-a02c-d7d821341eea)

(See #4133 for a discussion of this issue.)

I can fix the spacing with a show rule but that affects it everywhere:

```
a?b $(1, ?) "x?y"$

#show sym.quest: it => math.class("normal", it)

a?b $(1, ?) "x?y"$
```

![image](https://github.com/typst/typst/assets/2412819/41740b92-f0dc-479d-94b3-89b64fb445d4)

(It's not obvious in the image but the position of the first `?` in plain text is also subtly changed.)

Another use case was discussed in #2273: `#show "L": math.cal`, with the proposed workarounds having similar limitations.

**Comment by PgBiel** [comment]
The solution presented at https://github.com/typst/typst/discussions/2273#discussioncomment-7157366 would be the way to go, but fully fixing the pitfall at https://github.com/typst/typst/discussions/2273#discussioncomment-7791001 would require solving #1125

**Comment by knuesel** [comment]
In that example:

```
#show math.equation: it =>{
  show "L" : math.cal
  it
}

$
L "is the Lagrangian."
$```

I don't think #1125 would solve it? As I understand, once #1125 is done the text part will use the normal text font, but it will still be in the scope of the equation so it would still be subject to the show rule.

I guess it might go unnoticed if `math.cal` happens to work on normal text without any effect, but e.g. showing the Lagrangian symbol in red for example would also affect the text?

**Comment by knuesel** [comment]
Ah I see. Would that also apply to the question mark?

**Comment by PgBiel** [comment]
Outside of `"quotes"`, yes, likely.

**Comment by Enivex** [comment]
Duplicate of https://github.com/typst/typst/issues/254
